### DISCUSSION TITLE
Welcome to sumo Discussions!

### DISCUSSION THREAD

**Discussion by utf** [SMTG-UCL/sumo#128] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by subhadipniser333** [comment]
Hey How can I give a dotted line label to the band plot in sumo-bandplot . anyone can help?

**Reply by ajjackson** [reply]
This is not currently available, but has been requested a few times including this merge request https://github.com/SMTG-UCL/sumo/pull/68 where we reached some agreement on how those options could work. I'm sorry that we haven't got this into Sumo yet.

As a temporary hack, you could add the code
`    plt.axhline(0, 0, 1, linestyle='--', color='k')`
to line 431 of *sumo/cli/bandplot.py*. This will draw a dashed black horizontal line at zero. The code needs to be indented by four spaces to bring it in-line with the following block (`if save_files: ...`).

In Sumo, zero will be the VBM for a semiconductor. We do not consider the Fermi energy of band structure calculations to be reliable, so the VBM is a more useful reference point. (And makes it easy to read the bandgap!)

**Comment by subhadipniser333** [comment]
Hey Adam,
                Thanks for your comment. I got a dashed line at zero. Can
you help me with one more thing. is there any tool in sumo which can give
me a band structure with showing contribution from bulk and surface atoms
with different colour.

Thanks,


On Wed, May 5, 2021 at 11:54 PM Adam J. Jackson ***@***.***>
wrote:

> This is not currently available, but has been requested a few times
> including this merge request #68
> <https://github.com/SMTG-UCL/sumo/pull/68> where we reached some
> agreement on how those options could work. I'm sorry that we haven't got
> this into Sumo yet.
>
> As a temporary hack, you could add the code
> plt.axhline(0, 0, 1, linestyle='--', color='k')
> to line 431 of *sumo/cli/bandplot.py*. This will draw a dashed black
> horizontal line at zero.
>
> In Sumo, zero will be the VBM for a semiconductor. We do not consider the
> Fermi energy of band structure calculations to be reliable, so the VBM is a
> more useful reference point. (And makes it easy to read the bandgap!)
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/SMTG-UCL/sumo/discussions/128#discussioncomment-700738>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT6WIKU3NQQCYIMKOQBPRDDTMGEMRANCNFSM4VAVT3CA>
> .
>


-- 
*SUBHADIP PRADHAN*
RESEARCH SCHOLAR
SCHOOL OF PHYSICAL SCIENCE
NISER, BHUBANESWAR
ODISHA-752050, INDIA

**Reply by ajjackson** [reply]
I don't _think_ this is possible right now but it might be doable through some abuse of the `--atoms` option. @utf any ideas?

I would expect that rather than a band structure (which will be quite different for a slab than for the corresponding bulk material) the most appropriate analysis method would be an LDOS where the density of states is plotted against a spatial coordinate through the slab and into the vacuum. I'm afraid I'm not familiar with any specific existing tool for doing this with VASP.

**Reply by utf** [reply]
Yeah, this is currently possible for a density of states using the `--atom` option but not for a band structure.

**Comment by subhadipniser333** [comment]
Thanks  Adam and Alex for your reply.

On Sat, May 8, 2021 at 1:54 AM Alex Ganose ***@***.***> wrote:

> Yeah, this is currently possible for a density of states using the --atom
> option but not for a band structure.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/SMTG-UCL/sumo/discussions/128#discussioncomment-711532>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT6WIKVULILBYATITWCKZ53TMREAHANCNFSM4VAVT3CA>
> .
>


-- 
*SUBHADIP PRADHAN*
RESEARCH SCHOLAR
SCHOOL OF PHYSICAL SCIENCE
NISER, BHUBANESWAR
ODISHA-752050, INDIA
